spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RetryableRibbonLoadBalancingHttpClient.java

```diff
@@ -132,6 +132,11 @@ public RibbonApacheHttpResponse doWithRetry(RetryContext context) throws Excepti
 							.port(service.getPort()).path(newRequest.getURI().getPath())
 							.query(newRequest.getURI().getQuery()).fragment(newRequest.getURI().getFragment())
 							.build().encode().toUri());
+					
+					if (service instanceof RibbonServer) {
+						RibbonLoadBalancerContext rctx = new RibbonLoadBalancerContext(getLoadBalancer());
```

Comment:
Shouldnt we be using the bean here rather than creating a new instance?

Reply:
yes using the bean is better.  I have updated the code accordingly.
thanks!